### ISSUE TITLE
Error in `group_order` variable of group_XXXXXX_plot

### ISSUE THREAD

**Issue by jeremymiller** [AllenInstitute/scrattch.vis#17]
Hi @hypercompetent ,

I'm running into an issue where I can plot `group_violin_plot`s in the default order, but if I try and do it in any other order I get an error:

This works:
```
tome = "\\\\allen/programs/celltypes/workgroups/rnaseqanalysis/shiny/tomes/facs/human_MTG_bioRxiv/MTG_all.tome"
plotGenes <- c("AQP4","CUX2","GAD1")
anno      <- read_tome_anno(tome)
plotData  <- read_tome_gene_data(tome = tome, genes = plotGenes, regions = "both", units = "cpm")
scrattch.vis::group_violin_plot(data = plotData, anno = anno, genes = plotGenes, grouping = "cluster", 
  log_scale=FALSE, label_height = 50)
```
This doesn't:
```
scrattch.vis::group_violin_plot(data = plotData, anno = anno, genes = plotGenes, grouping = "cluster", 
  group_order = head(unique(anno$cluster_id)), log_scale=FALSE, label_height = 50)
Error in if (sum(plot_data[[gene]] == 0) == nrow(plot_data)) { : 
  missing value where TRUE/FALSE needed
```
group_box_plot and group_quasirandom_plot have the same error. 

group_dot_plot sort of works:
```
scrattch.vis::group_dot_plot(data = plotData, anno = anno, genes = plotGenes, grouping = "cluster", group_order = head(unique(anno$cluster_id)), log_scale=FALSE, label_height = 50)
Warning messages:
1: Removed 69 rows containing missing values (geom_point). 
2: Removed 69 rows containing missing values (geom_point). 
3: Removed 69 rows containing missing values (geom_point). 
4: Removed 69 rows containing missing values (geom_rect). 
5: Removed 69 rows containing missing values (geom_text). 
6: Removed 69 rows containing missing values (geom_text).
```
group_heatmap_plot has the same warnings and also looks weird

Hopefully it is either an easy fix, or me doing something dumb.

**Comment by hypercompetent** [comment]
This should now be fixed in the dev branch in pull #18 . If you can give it a test to make sure, I'll merge it in to master to get this resolved there, too.

-Lucas

**Comment by jeremymiller** [comment]
It works for me!  Go ahead and push the change.
